astropy/io/registry.py

```diff
@@ -321,9 +322,9 @@ def read(cls, *args, **kwargs):
             if len(args):
                 if isinstance(args[0], PATH_TYPES):
                     from ..utils.data import get_readable_fileobj
-                    # For Py3 the path might be a pathlib.Path object, so coerce
-                    # to a regular string.
-                    if six.PY3:
+                    # path might be a pathlib.Path object if HAS_PATHLIB,
+                    # so coerce to a regular string.
+                    if HAS_PATHLIB:
```

Comment:
@taldcroft If I understand correctly, you meant to say for example a file named ``x©.fits`` on Python 2 with the backport of pathlib independently installed would be a problem?
I cant understand why would it be a problem? 
I also tested it on my system on ``Python 2 `` and ``Python 3`` where it works correctly.